Patterns/LayeredProject.md

```diff
@@ -0,0 +1,100 @@
+
+### Project folders and namespaces
+
+Folders within projects should group similar objects (to a certain degree), and should be named what seems logical. Group converters in `Converters` and controllers in `Controllers`. Add more layers of folders when the folder becomes too big, like adding `Request` and `Response` folders in `Models` to separate request and response models. Group interfaces in `Abstractions`. Group extension (methods) in `Extensions`.
+
+The namespace of a file should always be comprised of its compound folder name, like `Contoso.CalendarHub.Api.Models.Request`.
+
+### Example
+
+- Contoso.CalendarHub.sln
+  - Contoso.CalendarHub.Api.csproj
+    - `Controllers`
+    - `JsonConverters`
+    - `Models`
+        - `Request`
+        - `Response`
+  - Contoso.CalendarHub.Api.Tests.csproj
+    - `LoginControllerTests`
+    - `UserControllerTests`
+  - Contoso.CalendarHub.Business.csproj
+    - `Abstractions`
+        - `Calendars`
+        - `Users`
+    - `Services`
+        - `Calendar`
+        - `User`
+  - Contoso.CalendarHub.Business.Tests.csproj
+    - `UserServiceTests`
+    - `CalendarServiceTest`
+  - Contoso.CalendarHub.Core.csproj
+    - `Attributes`
+    - `Extensions`
+    - `Enums`
+    - `Models`
+    - `Resolvers`
+  - Contoso.CalendarHub.Cms.csproj
+  - Contoso.CalendarHub.Data.Users.csproj
+  - Contoso.CalendarHub.Data.Invitations.csproj
+  - Contoso.CalendarHub.Functions.csproj
+    - `Attributes`
+    - `Constants`
+    - `Extensions`
+  - Contoso.CalendarHub.Gateway.Adyen.csproj
+    - `Models`
+  - Contoso.CalendarHub.Gateway.GoogleCalendar.csproj
+    - `Models`
+    - `Helpers`
+  - Contoso.CalendarHub.Gateway.GoogleCalendar.Tests.csproj
+  - Contoso.CalendarHub.Gateway.OutlookCalendar.csproj
+    - `Models`
+  - Contoso.CalendarHub.Gateway.OutlookCalendar.Tests.csproj
+
+## Inter-project data exchange and isolation
+
+When dividing a solution into several distinct projects it is important to maintain a level of isolation between these projects. Keeping these projects reasonably isolated helps to control the impact of changes; a change is one project should not force a lot of changes in other projects. Isolation will prevent developers from having to notify users of the API of your solution that some outbound data changes due to modification in a database column. Keep in mind to not overdo this, implementation details will always tend to spread a little bit to other parts of the solution.
+
+### Isolate models
+
+Models, or any object containing data, are usually passed around between various parts of the application. Each project should use its own type of object for its own use.
+
+- An API always has its own Request- and ResponseModels. These models describe the exact shape of requests it expects and responses it generates.
+- The Core projects contains data transfer objects (DTOs) which facilitate communication between parts of the application.
```

Comment:
As discussed previously, only Gateways and possibly Responses should be labeled as DTOs. The Core or Business layer usually contains Domain models.

Data Transfer Object is as the name implies, just a format of transporting the needed data to another application / system. It does so in the most compact / efficient way possible.

Reply:
I see it's been updated correctly in other parts of this manual 👍

Reply:
I find (also in examples online) that DTOs are generally used to communicate between things like Repositories and Services, and those are not send over the network.

Reply:
But I'll update and elaborate.